libazureinit/src/media.rs

```diff
@@ -76,19 +78,30 @@ pub const PATH_MOUNT_POINT: &str = "/run/azure-init/media/";
 const CDROM_VALID_FS: &[&str] = &["iso9660", "udf"];
 
 // Get a mounted device with any filesystem for CDROM
-pub fn get_mount_device() -> Result<Vec<String>, Error> {
-    let mut list_devices: Vec<String> = Vec::new();
-
-    while let Some(device) = block_utils::get_mounted_devices()?
+pub fn get_mount_device<F>(get_devices: F) -> Result<Vec<String>, Error>
+where
+    F: Fn() -> Result<Vec<Device>, Error>,
+{
+    let devices = get_devices()?;
+    let list_devices: Vec<String> = devices
         .into_iter()
-        .find(|dev| CDROM_VALID_FS.contains(&dev.fs_type.to_str()))
-    {
-        list_devices.push(device.name);
-    }
+        .filter_map(|dev| {
+            if CDROM_VALID_FS.contains(&dev.fs_type.to_str()) {
+                Some(dev.name)
+            } else {
+                None
+            }
+        })
+        .collect();
 
     Ok(list_devices)
 }
 
+// Wrapper function
+pub fn get_wrapped_mount_devices() -> Result<Vec<String>, Error> {
```

Comment:
Based on some side-channel conversations this is probably going away so this comment is really just a tip for the future.

One of the nice things about `pub` is that you can hide your sins from the outside world. What you could do is define this as `pub fn get_mount_device() ...` and then the function you're wrapping as `fn wrapped_get_mount_device(...)`. That way no one outside the library is even aware there's a wrapper layer.

Reply:
Another option would be to simply define only `pub fn get_mount_device(path: Option<&Path>) -> Result<Vec<String>, Error>`, without a wrapper layer. Then caller could either pass a custom path with `Some` or simply `None` to use the default.
IMHO code here fortunately does not need another wrapper layer like that.

```Rust
pub fn get_mount_device(path: Option<&Path>) -> Result<Vec<String>, Error> {
    // Create a new FsTab instance and parse the fstab file
    let fstab = FsTab::new(path.unwrap_or(Path::new(MTAB_PATH)));
```

Reply:
Should have resolved this, thank you!